sdk/identity/azure-identity/src/token_credential_impl.cpp

```diff
@@ -274,77 +277,80 @@ AccessToken TokenCredentialImpl::ParseToken(
-                 // 'expires_on' as RFC3339 date string (absolute timestamp)
-                 return DateTime::Parse(s, DateTime::DateFormat::Rfc3339);
-               }),
-               std::function<DateTime(std::string const&)>([](auto const& s) {
-                 // 'expires_on' as numeric string (posix time representing an absolute timestamp)
-                 return PosixTimeConverter::PosixTimeToDateTime(
-                     ParseNumericExpiration(s, MaxPosixTimestamp));
-               }),
-               std::function<DateTime(std::string const&)>([](auto const& s) {
-                 // 'expires_on' as RFC1123 date string (absolute timestamp)
-                 return DateTime::Parse(s, DateTime::DateFormat::Rfc1123);
-               }),
-           })
+      if (expiresOn.is_number_unsigned())
       {
         try
         {
-          accessToken.ExpiresOn = parse(expiresOnAsString);
-          return accessToken;
+          // 'expires_on' as number (posix time representing an absolute timestamp)
+          auto const value = expiresOn.get<std::int64_t>();
+          if (value <= MaxPosixTimestamp)
+          {
+            accessToken.ExpiresOn = PosixTimeConverter::PosixTimeToDateTime(value);
+            return accessToken;
+          }
         }
         catch (std::exception const&)
         {
-          // parse() has thrown, we may throw later.
+          // expiresIn.get<std::int64_t>() has thrown, we may throw later.
+        }
+      }
+
+      if (expiresOn.is_string())
+      {
+        auto const expiresOnAsString = expiresOn.get<std::string>();
+        for (auto const& parse : {
+                 std::function<DateTime(std::string const&)>([](auto const& s) {
+                   // 'expires_on' as RFC3339 date string (absolute timestamp)
+                   return DateTime::Parse(s, DateTime::DateFormat::Rfc3339);
+                 }),
+                 std::function<DateTime(std::string const&)>([](auto const& s) {
+                   // 'expires_on' as numeric string (posix time representing an absolute timestamp)
+                   return PosixTimeConverter::PosixTimeToDateTime(
+                       ParseNumericExpiration(s, MaxPosixTimestamp));
+                 }),
+                 std::function<DateTime(std::string const&)>([](auto const& s) {
+                   // 'expires_on' as RFC1123 date string (absolute timestamp)
+                   return DateTime::Parse(s, DateTime::DateFormat::Rfc1123);
+                 }),
+             })
+        {
+          try
+          {
+            accessToken.ExpiresOn = parse(expiresOnAsString);
+            return accessToken;
+          }
+          catch (std::exception const&)
+          {
+            // parse() has thrown, we may throw later.
+          }
         }
       }
     }
   }
 
   ThrowJsonPropertyError(
-      expiresOnPropertyName,
+      expiresOnPropertyNames.back(),
```

Comment:
What happens if the vector is empty, or is it a precondition of this function that it never will be (or at least never should be)? Do we always expect the last one to be the one that failed at this point?

Reply:
Line 280 would throw and we will not get here.